grakn-graql-shell/src/main/java/ai/grakn/graql/JsonSession.java

```diff
@@ -52,7 +52,11 @@
 
     private final Session session;
 
-    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(runnable -> {
+        Thread thread = Executors.defaultThreadFactory().newThread(runnable);
+        thread.setDaemon(true);
+        return thread;
+    });
```

Comment:
Possibly, but I doubt such a test would be actually robust. After all, various other things may be incidentally running threads at the same moment (such as if we were running tests in parallel).